contracts/token/ERC7984/extensions/ERC7984Omnibus.sol

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.27;
+
+import {FHE, euint64, externalEuint64, externalEaddress, eaddress} from "@fhevm/solidity/lib/FHE.sol";
+import {ERC7984} from "../ERC7984.sol";
+
+/**
+ * @dev Extension of {ERC7984} that emits additional events for omnibus transfers.
+ * These events contain encrypted addresses for the sub-account sender and recipient.
+ *
+ * NOTE: There is no onchain accounting for sub-accounts--integrators must track sub-account
+ * balances externally.
+ */
+abstract contract ERC7984Omnibus is ERC7984 {
+    /**
+     * @dev Emitted when a confidential transfer is made representing the onchain settlement of
+     * an omnibus transfer from `sender` to `recipient` of amount `amount`. Settlement occurs between
+     * `omnibusFrom` and `omnibusTo` and is represented in a matching {IERC7984-ConfidentialTransfer} event.
+     *
+     * NOTE: `omnibusFrom` and `omnibusTo` get permanent ACL allowances for `sender` and `recipient`.
+     */
+    event OmnibusConfidentialTransfer(
+        address indexed omnibusFrom,
+        address indexed omnibusTo,
+        eaddress sender,
+        eaddress indexed recipient,
+        euint64 amount
+    );
+
+    function confidentialTransferOmnibus(
+        address omnibusTo,
+        externalEaddress externalSender,
+        externalEaddress externalRecipient,
+        externalEuint64 externalAmount,
+        bytes calldata inputProof
+    ) public virtual returns (euint64) {
+        eaddress sender = FHE.fromExternal(externalSender, inputProof);
+        eaddress recipient = FHE.fromExternal(externalRecipient, inputProof);
+        euint64 amount = FHE.fromExternal(externalAmount, inputProof);
+
+        return confidentialTransferFromOmnibus(msg.sender, omnibusTo, sender, recipient, amount);
```

Comment:
Some allowances are given to `omnibusFrom` in `confidentialTransferFromOmnibus(..)` but it already has access to them.